nomt/src/cursor.rs

```diff
@@ -229,8 +279,121 @@ impl PageCacheCursor {
             })
     }
 
-    /// Modify the node at the current location of the cursor.
-    pub fn modify(&mut self, node: Node) {
+    fn read_leaf_children(&self) -> trie::LeafData {
+        let fetched_page;
+        let (page, (left_idx, right_idx)) = match self.cached_page {
+            None => {
+                fetched_page = self.pages.retrieve_sync(ROOT_PAGE_ID);
+                (&fetched_page, (0, 1))
+            }
+            Some((page_id, ref page)) => {
+                let path = last_page_path(&self.path, self.depth);
+                match child_node_indices(path) {
+                    Some((left, right)) => (page, (left, right)),
+                    None => {
+                        let child_page_idx = path.load_be::<u8>();
+
+                        // unwrap: this is true even for the leaf node placement, as less than the
+                        // entire 42nd page is used with 256 bit keys.
+                        let child_page_id = page_id.child_page_id(child_page_idx).expect(
+                            "Child index is 6 bits and Pages do not go deeper than the maximum layer, 42"
```

Comment:
Is that an adequate proof? It sounds like it doesn't go deeper than the layer 42, but `child_page_id` requires it to be strictly below 42. Ditto below and above.